scripts/runPostCrab.py

```diff
@@ -0,0 +1,219 @@
+    FWHash, FWBranch, FWUrl = getGitTagBranchUrl( os.path.join(CMSSW_BASE, 'src/cp3_llbb/Framework') )
+    # then the version of the analyzer
+    AnaHash, AnaBranch, AnaUrl = getGitTagBranchUrl( os.path.dirname( psetName ) )
+
+    print "##### Figure out the number of selected events"
+    # Need to get this from the output log of the jobs, and sum them all
+#    log_files = {'lfn': ['/store/user/obondu/GluGluToRadionToHHTo2B2VTo2L2Nu_M-260_narrow_13TeV-madgraph/GluGluToRadionToHHTo2B2VTo2L2Nu_M-260_narrow_Asympt25ns/150728_092137/0000/log/cmsRun_1.log.tar.gz'], 'pfn': ['srm://ingrid-se02.cism.ucl.ac.be:8444/srm/managerv2?SFN=/storage/data/cms/store/user/obondu/GluGluToRadionToHHTo2B2VTo2L2Nu_M-260_narrow_13TeV-madgraph/GluGluToRadionToHHTo2B2VTo2L2Nu_M-260_narrow_Asympt25ns/150728_092137/0000/log/cmsRun_1.log.tar.gz']}
+    nselected = 0
+    for lfn in log_files['lfn']:
+        tarLog =  ingridStoragePrefix + lfn
+        with tarfile.open(tarLog) as tar:
+            for tarFile in tar.getmembers():
+                if 'stdout' not in tarFile.name:
+                    continue
+                # For some reason, even though we are using python 2.7, the with statement here seems broken... Using contextlib to handle the file opening / reading cleanly
+                with contextlib.closing(tar.extractfile(tarFile)) as file:
+                    for line in file:
+                        if 'processed' not in line and 'selected' not in line:
+                            continue
+                        l = line.split()
+                        nselected  += int(line.split()[3])
+#    print nselected
+
+    print "##### For the path, check that the files there do correspond EXACTLY to the list of output files from crab"
+    # (crab being crab, we're never too careful!)
+#    output_files = {'lfn': ['/store/user/obondu/GluGluToRadionToHHTo2B2VTo2L2Nu_M-260_narrow_13TeV-madgraph/GluGluToRadionToHHTo2B2VTo2L2Nu_M-260_narrow_Asympt25ns/150728_092137/0000/output_mc_1.root'], 'pfn': ['srm://ingrid-se02.cism.ucl.ac.be:8444/srm/managerv2?SFN=/storage/data/cms/store/user/obondu/GluGluToRadionToHHTo2B2VTo2L2Nu_M-260_narrow_13TeV-madgraph/GluGluToRadionToHHTo2B2VTo2L2Nu_M-260_narrow_Asympt25ns/150728_092137/0000/output_mc_1.root']}
+    # first check what we do have locally
+    p = os.path.dirname( output_files['lfn'][0] )
+    localpath = ingridStoragePrefix + p
+    localfiles = [ os.path.join(p, f) for f in os.listdir(localpath) if os.path.isfile( os.path.join(localpath, f) ) and 'root' in f ]
+    # unordered comparison: the two list should be equal
+    if set(localfiles) != set(output_files['lfn']):
+        print "ERROR: the content of the path and the list of crab outputs are different, abort now!"
+        print "localfiles (what's here locally)= ", localfiles
+        print "outputfiles (what crab is saying)= ", output_files['lfn']
+        raise AssertionError("CRAB3 and local list of files do not match")
```

Comment:
Why do you do that? You can imagine some cases where you have duplicate files into your output folder, but you don't care, since you use the list of output files given by crab instead of listing the files in the output directory.

IMO this is not correct.